h2o-py/h2o/frame.py

```diff
@@ -139,7 +140,7 @@ def _upload_python_object(self, python_obj, destination_frame=None, header=0, se
 
         # create a temporary file that will be written to
         tmp_handle, tmp_path = tempfile.mkstemp(suffix=".csv")
-        tmp_file = os.fdopen(tmp_handle, 'w')
+        tmp_file = os.fdopen(tmp_handle, 'w', encoding='utf-8') if sys.version_info >= (3, 5) else os.fdopen(tmp_handle, 'w')
```

Comment:
as we apparently have to stick with `os.fdopen` here due to Py2 encoding inconsistencies, please use `PY2` constant instead of `sys.version` as we should always do: much easier to search for usages then: we don't support Py3 < 3.5 anyway.
maybe a simpler way to avoid redundancies:
```python
# as a global var
__fdopen_kwargs = {} if PY2 else {'encoding': 'utf-8'}

# each time we need fdopen:
tmp_file = os.fdopen(tmp_handle, 'w', **__fdopen_kwargs)
```

Reply:
hopefully we don't write much on the filesystem, as the write support for encodings in Py2 is a total disaster as it doesn't automatically convert `str` to proper strings (ie. `unicode`), hence this need to use `fdopen` apparently as it seems to write without any check/transformation (returned file object is not a `TextIOWrapper` in Py2).

Reply:
improved in 25e4ce9